### PR TITLE
Fixes #859

### LINKED ISSUE #859: Cannot read noColor of null. CLI Reporter > NodeJs Module
1. Newman Version (can be found via `newman -v`): > 3.3.1
 2. OS details (type, version, and architecture): MacOs
 3. Are you using Newman as a library, or via the CLI? Library -> NodeJS
 3. Did you encounter this recently, or has this bug always been there: Since 3.4.0
 4. Expected behaviour: Run Collection

This Issue seems to be regarding this commit: https://github.com/postmanlabs/newman/commit/acc7ddf6b7be5c191c045392334aeedd35d67eae

The Error thrown is
```
TypeError: Cannot read property 'noColor' of undefined
    at Object.inspector (~/node_modules/newman/lib/reporters/cli/cli-utils.js:109:37)
    at new PostmanCLIReporter (~/node_modules/newman/lib/reporters/cli/index.js:52:28)
```

After looking into this. It's first introduced with the above commit at Version 3.4.0. Version 3.3.1 works fine for me. When i log the runOptions (https://github.com/postmanlabs/newman/commit/acc7ddf6b7be5c191c045392334aeedd35d67eae#diff-7f96765916328a1d5ffea0fddaa42464R106) then they are null.

Thanks guys!

Edit: Maybe i should mention @kunagpal as Committer of this.

### PR DESCRIPTION
**Pull request by kunagpal**
Fixed colored output bug under library usage. Discovered by @derpoho in #859.

### CODE REVIEW

**Comment by shamasis** [comment]
@kunagpal please add a regression unit test that programmatic usage of Newman should work with minimum options.

**Comment by shamasis** [comment]
Tests for this is included in #861 hence merging.

File: test/library/export-environment.test.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            collection: 'test/fixtures/run/single-get-request.json',
+            environment: environment,
+            exportEnvironment: exportedEnvironmentPath
+        }, function (err) {
+            if (err) { return done(err); }
+
+            var environment;
+
+            try { environment = JSON.parse(fs.readFileSync(exportedEnvironmentPath).toString()); }
+            catch (e) { console.error(e); }
```
Comment (czardoz) [inline test/library/export-environment.test.js]: This should directly be returned to `done`

File: test/library/export-environment.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            collection: 'test/fixtures/run/single-request-failing.json',
+            environment: environment,
+            exportEnvironment: exportedEnvironmentPath
+        }, function (err) {
+            if (err) { return done(err); }
+
+            var environment;
+
+            try { environment = JSON.parse(fs.readFileSync(exportedEnvironmentPath).toString()); }
+            catch (e) { console.error(e); }
```
Comment (czardoz) [inline test/library/export-environment.test.js]: This too

**Comment by czardoz** [comment]
@kunagpal Minor comments, I'm merging because they're mostly inconsequential.
